### PR TITLE
feat(terraform): Definitions serialization with new definitions key/module objects

### PR DESCRIPTION
**Pull request by tronxd**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

- Define a custom serialization/deserialization of the new `TFDefinitionKey` and `TFModule` classes 

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by ChanochShayner — APPROVED** [review]

File: checkov/common/util/json_utils.py [hunk 1]
```diff
@@ -30,9 +34,62 @@ def default(self, o: Any) -> Any:
...
+    def encode_key(key):
+        if isinstance(key, TFDefinitionKey):
+            return str(key)
+        if isinstance(key, TFModule):
+            return str(key)
+        else:
+            return key
+
+
+def object_hook(dct):
```
Comment (bo156) [inline checkov/common/util/json_utils.py:63]: type annotations in this file?

File: checkov/common/util/json_utils.py [hunk 2]
```diff
@@ -30,9 +34,62 @@ def default(self, o: Any) -> Any:
...
+    try:
+        if dct is None:
+            return None
+        if isinstance(dct, dict):
+            dct_obj = deepcopy(dct)
+            if 'tf_source_modules' in dct and 'file_path' in dct:
+                return TFDefinitionKey(file_path=dct["file_path"],
+                                       tf_source_modules=object_hook(dct["tf_source_modules"]))
+            if 'path' in dct and 'name' in dct and 'foreach_idx' in dct and 'nested_tf_module' in dct:
```
Comment (bo156) [inline checkov/common/util/json_utils.py:72]: are any of this saved consts? if not you don't have to refactor them as they are used only once, but I figured they might be already defined

File: checkov/common/util/json_utils.py (hunk 1 above)
Comment (bo156) [inline checkov/common/util/json_utils.py:63]: can you rename this function to something a bit more clear? (or explain it to me if I'm missing something)

**Review by bo156 — APPROVED** [review]

File: checkov/common/util/json_utils.py (hunk 2 above)
Comment (tronxd) [reply checkov/common/util/json_utils.py:74]: Nope

File: checkov/common/util/json_utils.py (hunk 1 above)
Comment (tronxd) [reply checkov/common/util/json_utils.py]: object_hook is the standard way to pass a custom decoder to `json.loads`

**Pull request merged**